web/src/search_pill.ts

```diff
@@ -208,12 +208,11 @@ export function set_search_bar_contents(
         }
 
         if (user_pill_operators.has(term.operator) && term.operand !== "") {
-            const users = term.operand.split(",").map((email) => {
-                // This is definitely not undefined, because we just validated it
-                // with `Filter.is_valid_search_term`.
-                const user = people.get_by_email(email)!;
-                return user;
-            });
+            // This is definitely not undefined, because we just validated it
+            // with `Filter.is_valid_search_term`.
+            const users = people.get_users_from_ids(
+                people.user_ids_string_to_ids_array(term.operand),
+            );
```

Comment:
we cannot use `user_id` in search pills untill we use migrate to using `user_id` in user narrows, so removed that from this PR.